pkg/kubelet/volume/volume_manager.go

```diff
@@ -0,0 +1,357 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volumemanager
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/pod"
+	"k8s.io/kubernetes/pkg/kubelet/util/format"
+	"k8s.io/kubernetes/pkg/kubelet/volume/cache"
+	"k8s.io/kubernetes/pkg/kubelet/volume/populator"
+	"k8s.io/kubernetes/pkg/kubelet/volume/reconciler"
+	"k8s.io/kubernetes/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/util/operationexecutor"
+	"k8s.io/kubernetes/pkg/volume/util/types"
+	"k8s.io/kubernetes/pkg/volume/util/volumehelper"
+)
+
+const (
+	// reconcilerLoopSleepPeriod is the amount of time the reconciler loop waits
+	// between successive executions
+	reconcilerLoopSleepPeriod time.Duration = 100 * time.Millisecond
+
+	// desiredStateOfWorldPopulatorLoopSleepPeriod is the amount of time the
+	// DesiredStateOfWorldPopulator loop waits between successive executions
+	desiredStateOfWorldPopulatorLoopSleepPeriod time.Duration = 100 * time.Millisecond
+
+	// podAttachAndMountTimeout is the maximum amount of time the
+	// GetVolumesForPod call will wait for all volumes in the specified pod to
+	// be attached and mounted
+	podAttachAndMountTimeout time.Duration = 20 * time.Minute
+
+	// podAttachAndMountRetryInterval is the amount of time the GetVolumesForPod
+	// call waits before retrying
+	podAttachAndMountRetryInterval time.Duration = 300 * time.Millisecond
+
+	// waitForAttachTimeout is the maximum amount of time a
+	// operationexecutor.Mount call will wait for a volume to be attached.
+	waitForAttachTimeout time.Duration = 10 * time.Minute
```

Comment:
Add notes:

1.  This only blocks the goroutine that the sync function is running in for this pod
2.  The rationale for why this is so high